src/main/scala/nl/biopet/tools/pipelinestatus/Deps.scala

```diff
@@ -57,48 +58,138 @@ case class Deps(jobs: Map[String, Job], files: Array[JsObject])
   }
 
   def makePimRun(runId: String): Run = {
-    val links: List[Link] = this
-      .compressOnType()
-      .flatMap(x =>
-        x._2.map(y => Link("link", y, "output", x._1, "input", "test")))
-      .toList
+    def links: Array[Link] = this
+      .jobs.flatMap(x => x._2.dependsOnJobs.map(y => (x._1, y)))
+      .map(x =>
+        Link(
+          fromPort = if (jobs(x._2).configPath.nonEmpty)
+            "root" + jobs(x._2).configPath.mkString("/", "/", "/") + x._2 + "/output" else "root/" + x._2 + "/output",
+          toPort = if  (jobs(x._1).configPath.nonEmpty)
+            "root" + jobs(x._1).configPath.mkString("/", "/", "/") + x._1 + "/input" else "root/" + x._1 + "/input"))
+      .toArray
+
+    def jobsToNode(jobs: List[Job], depth: Int = 0): Array[Node] = {
+      val groups = jobs.groupBy(_.configPath.lift(depth))
+      (groups.filter(_._1.isEmpty).flatMap(_._2).map(j => Node(name = j.name, inPorts = Array(Port(name = "input")), outPorts = Array(Port(name = "output")))) ++
+        groups.filter(_._1.isDefined).map(g => Node(name = g._1.get, children = jobsToNode(g._2, depth + 1)))).toArray
```

Comment:
Will this be more clear if ._1 and ._2 are more explicit?
val meaningfulName = jobs(x._1) or somehting?

Reply:
not really but separating this will be more clear ;)